lib/plugins/aws/provider/awsProvider.js

```diff
@@ -428,7 +453,61 @@ class AwsProvider {
         },
         provider: {
           properties: {
-            apiGateway: { type: 'object', properties: { websocketApiId: { type: 'string' } } },
+            apiGateway: {
+              type: 'object',
+              properties: {
+                apiKeySourceType: {
+                  oneOf: ['HEADER', 'AUTHORIZER'].map(caseInsensitive),
+                },
+                binaryMediaTypes: {
+                  type: 'array',
+                  items: { type: 'string', pattern: '^\\S+\\/\\S+$' },
+                },
+                description: { type: 'string' },
+                metrics: { type: 'boolean' },
+                minimumCompressionSize: { type: 'integer', minimum: 0, maximum: 10485760 },
+                restApiId: { $ref: '#/definitions/awsCfInstruction' },
+                restApiResources: {
+                  anyOf: [
+                    {
+                      type: 'array',
+                      items: {
+                        type: 'object',
+                        properties: {
+                          path: { type: 'string' },
+                          resourceId: { type: 'string' },
+                        },
+                        required: [],
+                        additionalProperties: false,
+                      },
+                    },
+                    { type: 'object' },
+                  ],
+                },
+                restApiRootResourceId: { $ref: '#/definitions/awsCfInstruction' },
+                websocketApiId: { $ref: '#/definitions/awsCfInstruction' },
+              },
+              additionalProperties: false,
+            },
+            apiKeys: {
+              type: 'array',
+              items: {
+                oneOf: [
+                  { type: 'string' },
+                  {
+                    type: 'object',
+                    properties: {
+                      name: { type: 'string' },
+                      value: { type: 'string' },
+                      description: { type: 'string' },
+                      customerId: { type: 'string' },
+                    },
```

Comment:
We also support an object form where we accept one property (whatever named) with an array value (whatever string values)

```
provider:
  apiKeys:
    - free:
        - myFreeKey
        - ${opt:stage}-myFreeKey
    - paid:
        - myPaidKey
        - ${opt:stage}-myPaidKey
```

https://www.serverless.com/framework/docs/providers/aws/events/apigateway/#setting-api-keys-for-your-rest-api

(_I know it's a pretty quirky design_)